File: configlibrary/src/main/java/com/shmedo/configlibrary/iot/cmd/IOTCommandManager.java
package com.shmedo.configlibrary.iot.cmd;

import com.shmedo.configlibrary.ble.interfaces.Validater;
import com.shmedo.configlibrary.iot.enums.IOTCommandType;

/**
 * 创建者:   gonghe <br/>
 * 创建时间:  2020/11/11 <br/>
 * 描述：     TODO
 */
public class IOTCommandManager {

    private static final IOTCommandManager instance = new IOTCommandManager();

    public static IOTCommandManager getInstance() {
        return instance;
    }

    private IOTCommandManager() {
    }

    public String getCommand(IOTCommandType commandType) {
        IOTCommand cmd = new IOTCommand(commandType);
        return cmd.toString();
    }

    public <T extends Validater> String getCommand(IOTCommandType commandType, T parameter) {
        IOTCommand cmd = new IOTCommand(commandType, parameter);
        return cmd.toString();
    }


}

File: lib_device_base/src/main/java/com/shmedo/lib/device/base/iot_cmd/IOTCommandManager.kt
package com.shmedo.lib.device.base.iot_cmd

import com.shmedo.lib.device.base.iot_cmd.enums.IOTCommandType

/**
 * 创建者:   gonghe <br></br>
 * 创建时间:  2020/11/11 <br></br>
 * 描述：     TODO #gh#
 */
object IOTCommandManager {

    fun getCommand(commandType: IOTCommandType): String {
        return IOTCommand<Any>(commandType).toString()
    }

    fun <T> getCommand(commandType: IOTCommandType, parameter: T): String {
        return IOTCommand(commandType, parameter).toString()
    }

}